### ISSUE TITLE
getGIDSignInObject does not take in options we set in the Javascript

### ISSUE THREAD

**Issue by thagee** [EddyVerbruggen/cordova-plugin-googleplus#224]
Hi I don't know much about objective c but I do notice that getGIDSignInObject when i run my bebugger doesn't have any values stored in the options dictionary variable and there is no object found in the commands.arguments?  

I was wondering what would need to be changed to get this plugin to work as I need my app to notify google that my server would need access to communicate.

**Comment by sammuggPS** [comment]
Could you post a snippet of how you are passing in the options? Thanks!

**Comment by thagee** [comment]
I figured out why that was happening for some reason the webClientId was being passed as a function instead of a string.  I'm having another where if I sign out and sign back in I don't get a serverAuthCode given to me in the response and I can't use the previous server authcode.  Is there another option that I need to add or is this a bug on google's end?

**Comment by sammuggPS** [comment]
Glad you figured that part out.

serverAuthCode: Assuming you are using the latest code from GitHub, all I can say is make sure that you are passing in the webClientId **_and_** `offline: true`.

**Comment by thagee** [comment]
I am and it works the first time but on the second attempt  it doesn't go to the allow permission screen as it does for the facebook login.  Is there a option I need to turn on for the allow permission screen?

**Comment by sammuggPS** [comment]
Ah okay so there are two methods of logging out: `logout` and `disconnect`.

Logout basically says "Hey, this user won't be the default any more." This means that the current user won't be automatically logged in, allowing for other users who are signed in to the device (i.e. into Safari or have their account saved on Android) to have the opportunity to sign in. However, because Safari is where the list of users comes from, if you only have one account signed in to Safari, that account will be chosen automatically.

On the other hand, Disconnect will actually revoke the rights that are granted when "Allow" is clicked. Note that if you call disconnect, those effects are not always immediate. It could take a few minutes before the rights are actually revoked.

In short, the `logout` function won't get a permissions screen next time the same user logs in but the `disconnect` will.

**Comment by thagee** [comment]
Oh I see the library that is using this library doesn't use these function you have stated in their signout function.  Let me try both and see what happens.

Thanks for all the help :).
